### ISSUE TITLE
Cache exception handling broken in Quarkus 1.10.0.Final

### ISSUE THREAD

**Issue by knutwannheden** [quarkusio/quarkus#13399]
**Describe the bug**
When the cache value computation throws an exception a non-completed `CompletableFuture` value is written to the cache. The next time a client fetches the value for the same key from the cache the call will block forever in `CompletableFuture#get()` since the value was not removed from the cache in the exception handling.

For a simple reproducer see https://github.com/knutwannheden/caching-bug.

In Quarkus 1.10.0.CR1 this problem didn't exist and is thus probably caused by https://github.com/quarkusio/quarkus/commit/668ff4301ecd99c65bd720cc42d50c8104054709 which was integrated as part of the 1.10.0.Final release.

**Expected behavior**
An exception thrown by method annotated with `@CacheResult` should not have any impact on subsequent calls.

**Actual behavior**
An exception thrown by method annotated with `@CacheResult` causes subsequent calls to this method (for the same cache key) to block. Here is the corresponding stack trace for the blocking call:

```
"executor-thread-1@6796" daemon prio=5 tid=0x54 nid=NA waiting
java.lang.Thread.State: WAITING
at jdk.internal.misc.Unsafe.park(Unsafe.java:-1)
at java.util.concurrent.locks.LockSupport.park(LockSupport.java:194)
at java.util.concurrent.CompletableFuture$Signaller.block(CompletableFuture.java:1796)
at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3128)
at java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1823)
at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1998)
at io.quarkus.cache.runtime.CacheResultInterceptor.intercept(CacheResultInterceptor.java:48)
at io.quarkus.cache.runtime.CacheResultInterceptor_Bean.intercept(CacheResultInterceptor_Bean.zig:334)
at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
...
```

**To Reproduce**
Run unit test in https://github.com/knutwannheden/caching-bug.

1. git clone https://github.com/knutwannheden/caching-bug
2. cd caching-bug
3. ./mvnw verify

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Windows 10
 - Output of `java -version`: 11
 - Quarkus version or git rev: 1.10.0.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Maven 3.6.3

**Comment by gwenneg** [comment]
Thanks for reporting this @knutwannheden! I'll take care of it.

**Comment by knutwannheden** [comment]
@gwenneg Thanks for the prompt fix!

@gsmet Can this fix get backported to a 10.0.1 release?
